### PR TITLE
Backend for Dishes GetAll functionality

### PR DESCRIPTION
**Pull request by Soumya-Shreya**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -35,6 +36,7 @@ app.use(compression());
 
 // 6. Routes
 app.use("/api/ingredients", ingredients);
+app.use("/api/dishes", dishes)
```
Comment (SakshiShreya) [inline app.js:39]: add semi colon

File: models/dishes.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      type: String,
+      required: [true, "Dish should have a creator"],
+    },
+    updatedBy: {
+      type: String,
+      required: [true, "Dish should have a updater"],
+    },
+    recipe: {
+      type: String,
+      required: [false],
```
Comment (SakshiShreya) [inline models/dishes.js:19]: This line can be skipped if it is not required. By default it is false.

File: app.js (hunk 1 above)
Comment (Soumya-Shreya) [reply app.js]: Done

**Comment by Soumya-Shreya** [comment]
Data in DB:
image


GetAll API output:
image

File: controllers/dishes.js [hunk 3]
```diff
@@ -4,9 +4,10 @@ export async function getAllDishes(req, res, next) {
   try {
     const { page = 1, limit = 10, q = "" } = req.query;
     const filter = { name: { $regex: q, $options: "i" } };
-    const findPromise = Dishes.find(filter, {}, { skip: limit * (page - 1), limit }).populate(
-      "ingredients.ingredient",
-    );
+    const findPromise = Dishes.find(filter, {}, { skip: limit * (page - 1), limit })
+      .populate("ingredients", "ingredient")
```
Comment (Van-sh) [inline controllers/dishes.js:8]: this needs to be ingredients.ingredient

File: controllers/dishes.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    next(error);
+  }
+}
+
+export async function createDish(req, res, next) {
+  try {
+    const { body } = req;
+    // TODO: Update createdBy and updatedBy
+    body.createdBy = "Admin";
+    body.updatedBy = "Admin";
```
Comment (SakshiShreya) [inline controllers/dishes.js:24]: @Van-sh These will be set to current user now.

**Review by SakshiShreya — CHANGES_REQUESTED** [review]

File: controllers/dishes.js (hunk 4 above)
Comment (Soumya-Shreya) [reply controllers/dishes.js]: I updated them

**Pull request merged**